portal-impl/src/com/liferay/portal/events/ServicePreAction.java

```diff
@@ -675,14 +675,28 @@ private LayoutComposite _getViewableLayoutComposite(
 		List<Layout> accessibleLayouts = new ArrayList<>();
 
 		for (Layout curLayout : layouts) {
-			if ((ignoreHiddenLayouts || !curLayout.isHidden()) &&
-				_hasAccessPermission(permissionChecker, curLayout, false)) {
+			try {
+				if ((ignoreHiddenLayouts || !curLayout.isHidden()) &&
+					_hasAccessPermission(permissionChecker, curLayout, false)) {
 
-				if (accessibleLayouts.isEmpty() && !hasViewLayoutPermission) {
-					layout = curLayout;
+					if (accessibleLayouts.isEmpty() &&
+						!hasViewLayoutPermission) {
+
+						layout = curLayout;
+					}
+
+					accessibleLayouts.add(curLayout);
+				}
+			}
+			catch (Exception exception) {
+				if (_log.isWarnEnabled()) {
+					_log.warn(
+						"Error processing layout " + layout.getLayoutId());
 				}
 
-				accessibleLayouts.add(curLayout);
+				if (_log.isDebugEnabled()) {
```

Comment:
@manuelRives It is weird you have both Debug and Warn log for same exception, please remove the debug one here.